src/commands/deploy/deploy.js

```diff
@@ -462,6 +464,20 @@ const deploy = async (options, command) => {
   await command.authenticate(options.auth)
 
   let siteId = options.site || site.id
+
+  // replace site name by id if a match is found
+  if (options.site) {
+    const isSiteId = uuidValidate(options.site)
```

Comment:
AFAIK netlify apis creates uuidv4. Should we validate the version here too if the API is strict about it?

Reply:
Thanks for looking into this @tinfoil-knight.

I checked with our backend folks and it's better not to assume anything on the structure of the id.

How about we change [this line](https://github.com/netlify/cli/blob/9c8925222abc20aef50a600e83ae3ad6b7b0779d/src/commands/deploy/deploy.js#L468) to something like:
```
const [{ data: site, error: siteError }, sites] = await Promise.all([
        api
          .getSite({ siteId })
          .then((data) => ({ data }))
          .catch((error) => ({ error })),
        api.listSites({ name: siteId, filter: 'all', per_page: 1 }),
      ])
```

Then if `sites[0]` exists use that one, if not use the result of `getSite` (or error out)

Reply:
Yeah. This seems better. I'll make the change.